rust/src/dcerpc/dcerpc_udp.rs

```diff
@@ -0,0 +1,511 @@
+        None
+    }
+
+    pub fn handle_fragment_data(&mut self, input: &[u8], input_len: u16) -> u16 {
+        let mut retval: u16 = 0;
+        let hdrflags1 = self.get_hdr_flags1().unwrap_or(0);
+        let fraglenleft = self.fraglenleft;
+
+        // Update the stub params based on the packet type
+        match self.get_hdr_pkt_type().unwrap_or(0) {
+            DCERPC_TYPE_REQUEST => {
+                if let Some(ref mut req) = self.request {
+                    retval = evaluate_stub_params(
+                        input,
+                        input_len,
+                        hdrflags1,
+                        fraglenleft,
+                        &mut req.stub_data_buffer,
+                        &mut req.stub_data_buffer_len,
+                    );
+                }
+            }
+            DCERPC_TYPE_RESPONSE => {
+                if let Some(ref mut resp) = self.response {
+                    retval = evaluate_stub_params(
+                        input,
+                        input_len,
+                        hdrflags1,
+                        fraglenleft,
+                        &mut resp.stub_data_buffer,
+                        &mut resp.stub_data_buffer_len,
+                    );
+                }
+            }
+            _ => {
+                SCLogDebug!("Unrecognized packet type");
+                return 0;
+            }
+        }
+        // Update the remaining fragment length
+        self.fraglenleft -= retval;
+
+        retval
+    }
+
+    pub fn process_header(&mut self, input: &[u8]) -> i32 {
+        match parser::parse_dcerpc_udp_header(input) {
+            Ok((leftover_bytes, header)) => {
+                if header.rpc_vers != 4 {
+                    SCLogDebug!("DCERPC UDP Header did not validate.");
+                    return -1;
+                }
+                let mut uuidentry = DCERPCUuidEntry::new();
+                let auuid = header.activityuuid.to_vec();
+                uuidentry.uuid = auuid;
+                self.uuid_list.push(uuidentry);
+                self.header = Some(header);
+                (input.len() - leftover_bytes.len()) as i32
+            }
+            Err(nom::Err::Incomplete(_)) => {
+                // Insufficient data.
+                SCLogDebug!("Insufficient data while parsing DCERPC request");
+                -1
+            }
+            Err(_) => {
+                // Error, probably malformed data.
+                SCLogDebug!("An error occurred while parsing DCERPC request");
+                -1
+            }
+        }
+    }
+
+    pub fn handle_input_data(&mut self, input: &[u8]) -> AppLayerResult {
+        // Call header parser first
+        let mut parsed = self.process_header(input);
+        if parsed == -1 {
+            return AppLayerResult::err();
+        }
+
+        // Input left after successful header parsing should be 80 less
+        // than the original length
+        if input.len() < 80 {
```

Comment:
can `process_header` succeed if this check is true? If not, the check should probably be moved to before the call to `process_header`